From 8b9ad474e93667ea193e1ca9afdbdff6492fbe34 Mon Sep 17 00:00:00 2001
From: "Paul E. McKenney Subject: rcu: Remove" <paulmck@linux.vnet.ibm.com>
Date: Tue, 18 Jul 2017 14:54:45 +0100
Subject: [PATCH] exports from rcu_idle_exit() and rcu_idle_enter()

References: bnc#978907 Scheduler performance -- idle
Patch-mainline: v4.14-rc1
Git-commit: 16c0b106070f4760f7d1fffb0cd40393552b5294

The rcu_idle_exit() and rcu_idle_enter() functions are exported because they
were originally used by RCU_NONIDLE(), which was intended to be usable from
modules.  However, RCU_NONIDLE() now instead uses rcu_irq_enter_irqson()
and rcu_irq_exit_irqson(), which are not exported, and there have been
no complaints.

This commit therefore removes the exports from rcu_idle_exit() and
rcu_idle_enter().

Reported-by: Peter Zijlstra <peterz@infradead.org>
Signed-off-by: Paul E. McKenney <paulmck@linux.vnet.ibm.com>
Signed-off-by: Mel Gorman <mgorman@suse.de>
---
 kernel/rcu/tree.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/kernel/rcu/tree.c b/kernel/rcu/tree.c
index eef57e674f9b..6354641c2f8f 100644
--- a/kernel/rcu/tree.c
+++ b/kernel/rcu/tree.c
@@ -863,7 +863,6 @@ void rcu_idle_enter(void)
 	rcu_eqs_enter(false);
 	rcu_sysidle_enter(0);
 }
-EXPORT_SYMBOL_GPL(rcu_idle_enter);
 
 #ifdef CONFIG_NO_HZ_FULL
 /**
@@ -994,7 +993,6 @@ void rcu_idle_exit(void)
 	rcu_sysidle_exit(0);
 	local_irq_restore(flags);
 }
-EXPORT_SYMBOL_GPL(rcu_idle_exit);
 
 #ifdef CONFIG_NO_HZ_FULL
 /**
